src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -763,13 +763,46 @@ public override bool IsSingleByte
 
         public override Decoder GetDecoder()
         {
-            return new DecoderNLS(this);
+            return new ASCIIDecoder(this);
         }
 
-
         public override Encoder GetEncoder()
         {
-            return new EncoderNLS(this);
+            return new ASCIIEncoder(this);
+        }
```

Comment:
Checking; I don't think so yet. The GetEncoder call will be devirtualized; but then I think it needs to inline to then devirtualize at the next step; but I don't think inilining post devirtualization is there yet?